hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/container/ContainerImpl.java

```diff
@@ -1264,6 +1263,15 @@ public ContainerState transition(ContainerImpl container,
         return ContainerState.LOCALIZATION_FAILED;
       }
     }
+
+    static boolean isContainerRecoveredAsKilled(ContainerImpl container)
+    {
+      // container was killed but never launched
+      return container.recoveredAsKilled && (
```

Comment:
As we have a full function, it is probably to have a split sequence:
```
if (!container.recoveredAsKilled) {
    return false;
}
RecoveredContainerStatus containerStatus = container.recoveredStatus;
return containerStatus == RecoveredContainerStatus.REQUESTED || containerStatus  == RecoveredContainerStatus.QUEUED;
```

Reply:
Makes sense. Done.